adagrams/game.py

```diff
@@ -1,11 +1,126 @@
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    hand = random.sample(list(LETTER_POOL.keys()), 10)
+    return hand
 
+# def uses_available_letters(word, letter_bank):
+#     # Convert word to uppercase
+#     word = word.upper()
+#     # Create a counter of the letter_bank
+#     bank_counts = Counter(letter_bank)
+#     # Iterate over each letter in the word
+#     for letter in word:
+#         # If the letter is not in the bank_counts, or if the count is 0, return False
+#         if letter not in letter_bank or bank_counts[letter] == 0:
+#             return False
+#         # Otherwise, decrement the count of the letter in bank_counts
+#         bank_counts[letter] -= 1
+#     # If we make it through the loop without returning False, return True
+#     return True
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    word_counts = Counter(word)
+    for letter, count in word_counts.items():
+        if count > letter_bank.count(letter):
+            return False
+    return True
 
 def score_word(word):
-    pass
+    # word = word.upper()
+    # score = 0
+    # for letter in word:
+    #     if letter in "AEIOULNRST":
+    #         score += 1
+    #     elif letter in "DG":
+    #         score += 2
+    #     elif letter in "BCMP":
+    #         score += 3
+    #     elif letter in "FHVWY":
+    #         score += 4
+    #     elif letter == "K":
+    #         score += 5
+    #     elif letter in "JX":
+    #         score += 8
+    #     elif letter in "QZ":
+    #         score += 10
+    # if len(word) >= 7:
+    #     score += 8
+    # return score
+    scores = {
+        'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4,
+        'I': 1, 'J': 8, 'K': 5, 'L': 1, 'M': 3, 'N': 1, 'O': 1, 'P': 3,
+        'Q': 10, 'R': 1, 'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8,
+        'Y': 4, 'Z': 10}
+    score = sum(scores.get(letter, 0) for letter in word.upper())
+    if len(word) >= 7:
+        score += 8
+    return score
+
 
+
+# def get_highest_word_score(word_list):
+#     word_dict = {}
+#     for word in word_list:
+#         word_dict[word] = score_word(word)
+#     max_val = max(word_dict.values())
+#     max_word = max(word_dict, key=word_dict.get)
+#     for word, score in word_dict.items():
+#         if score == max_val:
+#             if len(max_word) == 10:
+#                 return max_word, max_val
+#             elif len(word) == 10 or len(word) < len(max_word):
+#                 max_word = word
+#     return max_word, max_val
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    # Calculate the score of each word in the list
+    word_scores = [(word, score_word(word)) for word in word_list]
+
+    # Find the maximum score among the list of words
+    max_score = max(word_scores, key=lambda x: x[1])[1]
```

Comment:
[nit] One edge case came to mind: what happens if `word_list` is empty?

Reply:
raised valueError